releasing.rst

```diff
@@ -29,124 +27,89 @@ we do a few things to prepare:
-
-6. Build documentation and review it for abnormalities.
+5. Build documentation and review it for abnormalities.
 
-7. Commit all changes from the above steps before moving on to the next stage.
+6. Commit all changes from the above steps.
 
-Note that any fixes done as a result of
-Stage 1 should be done through the normal process of making
-a pull request and going through review.
-However, from Stage 2 onward, the work is done directly
-on the ``master`` branch.
-Proceed with caution!
-
-Stage 2: Releasing
-==================
-
-Once everything is prepared, we're ready to do the release.
-It should be okay to work in the same directory that you
-do development, but if you want to be extra safe,
-you can do a fresh clone of the repository
-into a separate directory.
-
-1. Change the version information for your project.
+7. Change the version information for your project.
    For most Python projects, it lives in ``project/version.py``.
    See that file for details.
 
-2. Set the release date in the changelog (usually ``CHANGES.rst``).
-
-3. Run ``collective.checkdocs`` to ensure proper formatting for PyPI.
-
-   .. code-block:: bash
-
-      python setup.py checkdocs
+8. Set the release date in the changelog, ``CHANGES.rst``.
 
-4. ``git add`` the changes above and make a release commit with
+9. ``git add`` the changes above and make a release commit with
 
    .. code-block:: bash
 
       git commit -m "Release version X.Y.Z"
 
-5. Create release packages.
-   We build source distributions and wheels whenever possible.
+10. Push with ``git push origin release-candidate-X.Y.Z``
+    to start a TravisCI build that will run several checks
+    to ensure that the repository is in a good state for release.
 
-   .. code-block:: bash
+11. If the TravisCI build fails, fix the issues, commit them before the
+    release commit, and push again.
 
-      python setup.py sdist bdist_wheel
+12. When the TravisCI build succeeds,
+    inspect the release and associated metadata and files at
+    https://test.pypi.org/project/<project>/
 
-6. Review the release packages to ensure extra files
-   (like those in ``.ipynb_checkpoints``) are not included.
-   The size of the package is a good indication of whether
-   extra files are present.
+    In particular, make sure that extraneous files are not
+    included in the released files.
+    File sizes give a good indication of whether
+    extra files are present.
 
-7. Upload the release packages to PyPI.
+    If there are any issues, fix them and commit them before
+    the release commit.
 
-   .. code-block:: bash
+13. If any non-trivial changes were made in this process,
+    open a pull request on the ``release-candidate-X.Y.Z`` branch
+    to have the changes reviewed.
+
+Stage 2: Tagged release
+=======================
 
-      twine upload dist/<package-version>.tar.gz
-      twine upload dist/<package-version-extratags>.whl
+1. Merge the ``release-candidate-X.Y.Z`` branch into ``master``.
```

Comment:
I often do all this part in the `release-candidate` branch (and then merge at the end), just on the off chance that there's a mistake in e.g. step 3 or 4 which you'd want to force push to fix.

Reply:
Makes sense, will move the merge to the end!